velox/exec/WindowPartition.cpp

```diff
@@ -383,6 +386,14 @@ void WindowPartition::updateKRangeFrameBounds(
             orderByRowColumn.nullByte(),
             orderByRowColumn.nullMask()));
 
+    // Mark the frame invalid if the frame bound is NaN.
```

Comment:
My bad that we do null check before NaN check. I'm surprised that didn't get caught in the testing.

The NULL check happens in the next if condition that compares for the equality of RowContainer values.  Since NaN != NaN, we could move the unguarded NaN check in the else side of the condition below.

Reply:
Hi @aditi-pandit, I got what you mean. But it looks like we cannot avoid the check for NULL in RowContainer::isNanAt even in the else branch here? The reason is that RowContainer::compare returns non-zero if frame bound is NULL but order-by value is not NULL. Also, RowContainer::compare actually considers NaN == NaN.

Reply:
@kagamiori : I see. Guess we have to use the version of the NaN check with the NULL check. We can do the check before the DCHECK of the NULL column values at line 378. Thanks.

Reply:
Hi @aditi-pandit, I've updated the PR. Please take another look. Thank you!

Reply:
@kagamiori : Your code looks fine to me except for the test comment below. I'll approve after the test fix.